frontend/test/src/reducer/workspaceList.spec.js

```diff
@@ -0,0 +1,210 @@
+        expect(rez).to.deep.equal([
+          ...initialState,
+          {
+            ...serialize(firstWorkspaceFromApi, serializeWorkspaceListProps),
+            memberList: [{
+              ...serializeMember({
+                user: globalManagerFromApi,
+                do_notify: true,
+                role: ROLE.workspaceManager
+              })
+            }]
+          }
+        ])
+      })
+    })
+
+    describe(`${UPDATE}/${WORKSPACE_MEMBER}`, () => {
+      const rez = workspaceList([
+        ...initialState,
+        {
+          ...serialize(firstWorkspaceFromApi, serializeWorkspaceListProps),
+          memberList: [{
+            ...serializeMember({
+              user: globalManagerFromApi,
+              do_notify: true,
+              role: ROLE.workspaceManager
+            })
+          }]
+        }
+      ],
+      updateWorkspaceMember(globalManagerFromApi, firstWorkspaceFromApi.workspace_id, { ...globalManagerFromApi, do_notify: false, role: ROLE.contributor })
+      )
+
+      it('should return a workspace list with the member correctly updated in the right workspace', () => {
+        expect(rez).to.deep.equal([
+          ...initialState,
+          {
+            ...serialize(firstWorkspaceFromApi, serializeWorkspaceListProps),
+            memberList: [{
+              ...serializeMember({
+                user: globalManagerFromApi,
+                do_notify: false,
+                role: ROLE.contributor
+              })
+            }]
+          }
+        ])
+      })
+    })
+
+    describe(`${REMOVE}/${WORKSPACE_MEMBER}`, () => {
+      const rez = workspaceList([
+        ...initialState,
+        {
+          ...serialize(firstWorkspaceFromApi, serializeWorkspaceListProps),
+          memberList: [{
+            ...serializeMember({
+              user: globalManagerFromApi,
+              do_notify: true,
+              role: ROLE.workspaceManager
+            })
+          }]
+        }
+      ],
+      removeWorkspaceMember(globalManagerFromApi.user_id, firstWorkspaceFromApi.workspace_id)
+      )
+
+      it('should return a workspace list without the member removed in the right workspace', () => {
+        expect(rez).to.deep.equal([
+          ...initialState,
+          {
+            ...serialize(firstWorkspaceFromApi, serializeWorkspaceListProps),
+            memberList: []
+          }
+        ])
+      })
+    })
+
+    describe(`${SET}/${WORKSPACE_MEMBER}`, () => {
+      const rez = workspaceList([
+        {
+          ...serialize(firstWorkspaceFromApi, serializeWorkspaceListProps),
+          memberList: []
+        }
+      ],
+      setWorkspaceListMemberList([{
+        ...serialize(firstWorkspaceFromApi, serializeWorkspaceListProps),
+        workspaceId: firstWorkspaceFromApi.workspace_id,
+        memberList: [{ user: globalManagerFromApi, do_notify: false, role: ROLE.contributor }]
+      }]))
```

Comment:
This part isn't very readable
I tried this:
```suggestion
      const serializedFirstWorkspace = {
        serialize(firstWorkspaceFromApi, serializeWorkspaceListProps),
        memberList: []
      }

      const rez = workspaceList(
        [serializedFirstWorkspace],
        setWorkspaceListMemberList([{
          ...serializedFirstWorkspace,
          workspaceId: firstWorkspaceFromApi.workspace_id,
          memberList: [{ user: globalManagerFromApi, do_notify: false, role: ROLE.contributor }]
        }])
      )
```
Note that since you use a serialized version many times, you can create a const for it at the top of your tests.
You also have an exported serialized version of `firstWorkspaceFromApi` in the same fixture file